### PR TITLE
Move cart and order API logic from Storefront and target the frontend to this

### PR DESCRIPTION
**Pull request by trueboroda**
VP-1075
Move api logic from Storefront project for security and business logic moving to server side.
Target and debug the frontend  part to new API.

### CODE REVIEW

File: VirtoCommerce.JavaScriptShoppingCart.Data/VirtoCommerce.JavaScriptShoppingCart.Data.csproj [hunk 1]
```diff
@@ -36,6 +36,9 @@
     <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
       <HintPath>..\packages\Newtonsoft.Json.12.0.2\lib\net45\Newtonsoft.Json.dll</HintPath>
     </Reference>
+    <Reference Include="Omu.ValueInjecter, Version=2.3.3.0, Culture=neutral, PublicKeyToken=c7694541b0ac80e4, processorArchitecture=MSIL">
```
Comment (yecli) [inline VirtoCommerce.JavaScriptShoppingCart.Data/VirtoCommerce.JavaScriptShoppingCart.Data.csproj:39]: Where do you need ValueInjecter?
In this stage we could add "manual" converters.
Later we want to use Automapper.

File: VirtoCommerce.JavaScriptShoppingCart.Web/Controllers/Api/JsCartCartController.cs.orig [hunk 2]
```diff
@@ -0,0 +1,319 @@
+using System;
```
Comment (yecli) [inline VirtoCommerce.JavaScriptShoppingCart.Web/Controllers/Api/JsCartCartController.cs.orig:1]: Probably need to remove, it is temp merging file?

Comment (t13ka) [reply VirtoCommerce.JavaScriptShoppingCart.Web/Controllers/Api/JsCartCartController.cs.orig:1]: Agree. This looks like a temp file

Comment (trueboroda) [reply VirtoCommerce.JavaScriptShoppingCart.Web/Controllers/Api/JsCartCartController.cs.orig:1]: Yes, it is right.

File: VirtoCommerce.JavaScriptShoppingCart.Data/VirtoCommerce.JavaScriptShoppingCart.Data.csproj (hunk 1 above)
Comment (trueboroda) [reply VirtoCommerce.JavaScriptShoppingCart.Data/VirtoCommerce.JavaScriptShoppingCart.Data.csproj:39]: https://github.com/VirtoCommerce/vc-module-javascript-shoppingcart/blob/dd62a2a9ba5743c1ee939ada7198cc6ec284108d/VirtoCommerce.JavaScriptShoppingCart.Data/Converters/CartConverter.cs#L601
In this case, the conversion comes from the object coming in the base class parameter. In fact, an object can have the type of an inherited class from the base PromotionReward. 
That is why this ValueInjecter is used. I think you can replace it with AutoMapper.

File: VirtoCommerce.JavaScriptShoppingCart.Data/Converters/CartConverter.cs [hunk 3]
```diff
@@ -230,5 +520,265 @@ public static CartShipmentItem ToShipmentItem(this cartDto.ShipmentItem shipment
...
+                ProductId = lineItem.ProductId,
+                Discount = lineItem.DiscountTotal.Amount,
+
+                // Use only base price for discount evaluation
+                Price = lineItem.SalePrice.Amount,
+                Quantity = lineItem.Quantity,
+                InStockQuantity = lineItem.InStockQuantity,
+
+                // Outline = lineItem.Product.Outline,
+                Variations = null, // TODO
```
[the remark this answers is not in the document]
Comment (trueboroda) [reply VirtoCommerce.JavaScriptShoppingCart.Data/Converters/CartConverter.cs:532]: It is not my todo comment.

**Pull request merged**